cstar_ocean/input_dataset.py

```diff
@@ -193,41 +273,61 @@ def check_exists_locally(self, local_dir: str) -> bool:
         return self.exists_locally
 
 
-class ModelGrid(InputDataset):
+class ROMSInputDataset(InputDataset, ABC):
+    (
+        """
+    ROMS-specific implementation of `InputDataset` (doc below)
+
+    Extends `get()` method to generate dataset using roms-tools in the case that `source`
+    points to a yaml file.
+
+    Docstring for InputDataset:
+    ---------------------------
+    """
+    ) + (InputDataset.__doc__ or "")
```

Comment:
Are you trying to edit the docstring of the `ROMSInputDataset` class here? Because you can access `__doc__` directly, so I think you can do

```python
ROMS_DATASET_DOC = """..."""

class ROMSInputDataset(InputDataset, ABC):
    __doc__ = InputDataset.__doc__ + ROMS_DATASET_DOC
```

Reply:
Neat. I'm now remembering that I meant to ask you about how you'd go about this as PEP-257 isn't very clear. Almost all the information is contained in the `InputDataset` docstring _but_ the user will never actually be handling an `InputDataset` directly now it's an ABC, and we can imagine them basically only ever handling `ROMSInputDataset` types. It seems a bit daft to write a docstring that says "go look at this other docstring that you'd never otherwise encounter, this is basically the same except for the following differences". OTOH just dumping the contents of that parent docstring into this one (as I've done here) doesn't feel very clean either. Any ideas?

Reply:
I've seen this problem before but don't know of a neater solution.

> just dumping the contents of that parent docstring into this one (as I've done here) doesn't feel very clean either

This is probably fine imo.